### DISCUSSION TITLE
how could I limit size of the lsp documentation window?

### DISCUSSION THREAD

**Discussion by shinonomeow** [LazyVim/LazyVim#1851] [Q&A] [answered]
<img width="2150" alt="image" src="https://github.com/LazyVim/LazyVim/assets/67734180/9a7495dc-4a96-4d57-ae6e-9a6a46b62b3e">
The window is blocking the function I'm currently typing, and I don't know how to adjust its size. Thank you for your help.

i found some code 
```
local orig_util_open_floating_preview = vim.lsp.util.open_floating_preview
function vim.lsp.util.open_floating_preview(contents, syntax, opts, ...)
  opts = opts or {}
  opts.border = opts.border or 'single'
  opts.max_width= opts.max_width or 80
  return orig_util_open_floating_preview(contents, syntax, opts, ...)
end
```
but I don't how to use it

**Comment by dpetka2001** [comment]
You can try the following and see how it looks for you
```lua
  {
    "folke/noice.nvim",
    opts = {
      presets = {
        lsp_doc_border = true,
      },
    },
  },
```

**Reply by shinonomeow** [reply]
<img width="2150" alt="image" src="https://github.com/LazyVim/LazyVim/assets/67734180/21e19c33-e258-40e3-a7a8-8afc088b013a">
It looks like this.

**Reply by dpetka2001** [reply, marked as answer]
Try
```lua
  {
    "folke/noice.nvim",
    opts = {
      presets = {
        lsp_doc_border = true,
      },
      lsp = {
        signature = {
          opts = {
            size = {
              height = 10,
            },
          },
        },
      },
    },
  },
```
and play around with the value of `height` to make the height of the window smaller, so that it doesn't get in the way when you're typing. You can also disregard the preset about `lsp_doc_border` if you don't like rounded borders or just leave it like that.

**Reply by shinonomeow** [reply]
Thank you very much for your help. This configuration is very useful for me.
